src/database/sql_database.cc

```diff
@@ -1037,9 +1056,21 @@ std::vector<std::shared_ptr<CdsObject>> SQLDatabase::search(const SearchParam& p
             searchColumnMapper->mapQuoted(SearchCol::Flags), OBJECT_FLAG_SEARCHABLE, searchColumnMapper->mapQuoted(SearchCol::ObjectType), OBJECT_TYPE_CONTAINER));
     }
 
-    beginTransaction("search");
-    auto countSQL = fmt::format("SELECT COUNT(DISTINCT {}) FROM {} WHERE {}", searchColumnMapper->mapQuoted(UPNP_SEARCH_ID), sql_search_query, searchSQL);
+    bool rootContainer = param.getContainerID() == "0";
+
+    std::string countSQL;
+    if (rootContainer) {
+        // Use faster, non-recursive search for root container
+        countSQL = fmt::format("SELECT COUNT(DISTINCT {}) FROM {} WHERE {}", searchColumnMapper->mapQuoted(UPNP_SEARCH_ID), sql_search_query, searchSQL);
+    } else {
+        // Use recursive container search
+        const std::string countSelect = fmt::format("COUNT(DISTINCT {})", searchColumnMapper->mapQuoted(UPNP_SEARCH_ID));
+        countSQL = fmt::format(sql_search_container_query_format, param.getContainerID(), countSelect);
+        countSQL += fmt::format(" WHERE {}", searchSQL);
```

Comment:
If you add ` WHERE {} ` to `sql_search_container_query_format` you can merge the statements